app.js

```diff
@@ -80,12 +81,20 @@ for (let i = 0; i < 8; i++) {
                 if (moveLogic(currState, clickx, clicky, x, y)) {
                     currState[y][x] = currState[clicky][clickx]
                     currState[clicky][clickx] = 0
+                    if ((y === 0 || y === 7) && currState[y][x][0] === 'p') {
+                        currState[y][x] = 'q' + color
+                    }
                     whiteTurn = !whiteTurn
+                    color = whiteTurn ? 'w' : 'b'
```

Comment:
Do we really need the `color` to be global variable? It can be just a local one in this scope